docs/common/project/how-to-release.md

```diff
@@ -255,3 +260,63 @@ Java artifacts are available from Maven Central.
 
 Thanks to everyone for contributing!
 ```
+
+### Documentation Release
+
+Documentation needs to be updated as a part of Iceberg release after a release candidate is passed.
+The commands described below assume the `iceberg-docs` repository and `iceberg` repository are in the same parent directory locally, 
+and the release manager is executing commands in the `iceberg` repository.
+Adjust the commands accordingly if it is not the case.
+
+#### Release notes
+
+Release notes need to be drafted and published in the `iceberg` repository and reviewed by the community.
+
+#### Iceberg version update
+
+Before release, publish a PR in the `iceberg-docs` repository to update `landing-page/config.toml` and `docs/config.toml` to use the new Iceberg version number.
+
+#### Documentation update
+
+To start the release process, run the following steps in the `iceberg-docs` repository to copy docs over:
+
+```shell
+rm -rf ../iceberg-docs/docs/content/docs
+rm -rf ../iceberg-docs/landing-page/content/common
+cp -r docs/versioned ../iceberg-docs/docs/content/docs
```

Comment:
This is where we maintain `latest`?

Reply:
I think I get it. Since this is going to be the basis for the new branch, we put the content directly in the main branch, then branch for the version.